src/components/community/Boards/BoardSearch.tsx

```diff
@@ -7,29 +7,32 @@ import SectionTitle from '@/components/community/SectionTitle'
 import SearchBox from '@/components/timetable/SearchBox'
 import NoticeModal from '@/components/ui/modal/NoticeModal'
 import { COMMUNITY_SEARCH_MESSAGES } from '@/lib/messages/community'
+import { useQueryParams } from '@/util/hooks/useQueryParams'
 import useIntersect from '@/util/useIntersect'
 import { useModal } from '@/util/useModal'
-import { useSearch } from '@/util/useSearch'
 
 const boardConfig: { [key: string]: number } = {
   community: 1,
   question: 2,
   information: 3,
 }
+type SearchParams = {
+  keyword: string
+}
 const BoardSearch = () => {
   const { boardName } = useParams()
-  const { searchParam, handleSetParam, deleteParam } = useSearch()
+  const [queryParams, setQueryParams] = useQueryParams<SearchParams>()
+
   const { data, hasNextPage, isFetching, fetchNextPage } = useGetPostsByBoard(boardConfig[boardName ?? 'community'])
   const { isOpen, handleOpen } = useModal(true)
+
   const onSubmit = (searchParam: string) => {
-    if (searchParam === '') {
-      return deleteParam('keyword')
-    }
-    if (searchParam.length < 2) return handleOpen()
-    handleSetParam('keyword', searchParam)
+    if (searchParam.length < 2 && searchParam.length > 0) return handleOpen()
+    const keyword = !searchParam.length ? { keyword: undefined } : { keyword: searchParam }
+    setQueryParams(keyword as SearchParams)
```

Comment:
```suggestion
    setQueryParams({ keyword: searchParam.length ? searchParam : undefined })
```
이런식으로 간추리는 것도 괜찮은 거 같아요